features/example_groups/shared_examples.feature

```diff
@@ -19,6 +19,46 @@ Feature: shared examples
   anything special (like autoload). Doing so would require a strict naming
   convention for files that would break existing suites.
 
+  **WARNING:** When you include parameterized examples in the current context multiple
+  times, you are actually overriding previous `let` definitions and last declaration wins.
+  So if you have this kind of shared example (or shared context)
+
+  ```ruby
+  RSpec.shared_examples "some example" do |parameter|
+    let(:something) { parameter }
+    it "uses the given parameter" do
+      expect(something).to eq(parameter)
+    end
+  end
+
+  RSpec.describe SomeClass do
+    include_example "some example", "parameter1"
+    include_example "some example", "parameter2"
+  end
+  ```
+
+  You're actually doing this (notice that first example will fail):
+
+  ```ruby
+  RSpec.describe SomeClass do
+    \# Reordered code for better understanding of what is happening
+    let(:something) { "parameter1" }
+    let(:something) { "parameter2" }
+
+    it "uses the given parameter" do
+      \# This example will fail because last let "wins"
+      expect(something).to eq("parameter1")
+    end
+
+    it "uses the given parameter" do
+      expect(something).to eq("parameter2")
+    end
+  end
+  ```
+
+  To prevent this kind of subtle errors a warning is raised if you declare multiple
+  let blocks with the same name in the same context.
```

Comment:
Well done on the doc update...it definitely helps.  Do you think we should mention the solution being to use `it_behaves_like`, which uses nesting to avoid the issue?

Also, two minor nitpicks:

*  "warning is raised" is perhaps a bit confusing -- `raise` has the specific meaning in ruby of raising an exception.  Maybe "a warning is emitted" would avoid that confusion?
* Would be nice to put backticks around `let` so it renders as code on relish.